uvicorn/server.py

```diff
@@ -299,7 +339,7 @@ def install_signal_handlers(self) -> None:
                 signal.signal(sig, self.handle_exit)
 
     def handle_exit(self, sig: int, frame: Optional[FrameType]) -> None:
-
+        self._delayed_signal = signal.Signals(sig)
```

Comment:
What is the supposed behavior of multiple signals here? Because we only care about the last one...

Reply:
This just stores one signal here. `signal.Signals` is the enum of all signals, `signal.Signals(sig)` selects the one corresponding to `sig`.

Reply:
Yeah, I'm asking why it only stores one.

Reply:
Good question. I did not find anything on handling a sequence of signals, but it is important that [the "fatal" signal is propagated](https://www.cons.org/cracauer/sigint.html) – just handling the last does that.

According to [this SO resource](https://stackoverflow.com/questions/48434964/interruption-of-signal-handler-with-other-signal), at least the de-facto behaviour on Linux is for signal handlers to be interrupted by newer signals. That means if the currently handled signal does not kill the application, the previously invoked signal handler gets a chance. That suggests a LIFO handling of signals.

I will refactor this to storing all signals and replaying them in LIFO order.

Reply:
All captured signals are stored now and replayed in LIFO order.